### PR TITLE
Code cleanup, bug fixes, and comments in Survival Predictor

### PR DESCRIPTION
**Pull request by saimarathore**
Code cleanup and bug fixes in Survival Predictor. Also added comments in all the files associated with Survival Predictor.

### CODE REVIEW

File: src/applications/SurvivalPredictor.h [hunk 1]
```diff
@@ -861,115 +860,135 @@ std::vector<typename ImageType::Pointer> SurvivalPredictor::RevisedTumorArea(con
...
+    else if (atlasIt.Get() == 27)
+      roi21It.Set(317);
+    else if (atlasIt.Get() == 29)
+      roi21It.Set(318);
+    else if (atlasIt.Get() == 4)
+      roi21It.Set(319);
+    else if (atlasIt.Get() == 67 || atlasIt.Get() == 76)
+      roi21It.Set(320);
+    else if (atlasIt.Get() == 7)
+      roi21It.Set(321);
```
Comment (ashishsingh18) [inline src/applications/SurvivalPredictor.h:932]: What are the hard-coded values? Needs comments.

File: src/applications/SurvivalPredictor.cpp [hunk 2]
```diff
@@ -781,19 +649,19 @@ VectorDouble SurvivalPredictor::SurvivalPredictionOnExistingModel(const std::str
...
+  VariableSizeMatrixType SelectedFeatures; 
+  SelectedFeatures.SetSize(allfeatures.Rows(),indicesselectedfeatures.Size()+1);
   int counter = 0;
-  for (unsigned int i = 0; i < selectedfeatures.Size(); i++)
+  for (unsigned int i = 0; i < indicesselectedfeatures.Size(); i++)
   {
-    for (unsigned int j = 0; j < SixModelFeatures.Rows(); j++)
-      SixModelSelectedFeatures(j, counter) = SixModelFeatures(j, selectedfeatures[i]);
+    for (unsigned int j = 0; j < allfeatures.Rows(); j++)
+      SelectedFeatures(j, counter) = allfeatures(j, indicesselectedfeatures[i]);
```
[the remark this answers is not in the document]
Comment (saimarathore) [reply src/applications/SurvivalPredictor.cpp:662]: comments added

File: src/applications/SurvivalPredictor.h [hunk 3]
```diff
@@ -861,115 +860,135 @@ std::vector<typename ImageType::Pointer> SurvivalPredictor::RevisedTumorArea(con
...
+  roi9ImagePointer->SetRequestedRegion(atlaswarpedImagePointer->GetLargestPossibleRegion());
+  roi9ImagePointer->SetBufferedRegion(atlaswarpedImagePointer->GetBufferedRegion());
+  roi9ImagePointer->Allocate();
+  roi9ImagePointer->FillBuffer(0);
+
+  typedef itk::ImageRegionIteratorWithIndex <ImageType> IteratorType;
+  IteratorType roi9It(roi9ImagePointer, roi9ImagePointer->GetLargestPossibleRegion()); 
+  IteratorType roi21It(roi21ImagePointer, roi21ImagePointer->GetLargestPossibleRegion());
+  IteratorType atlasIt(atlaswarpedImagePointer, atlaswarpedImagePointer->GetLargestPossibleRegion());
+
```
[the remark this answers is not in the document]
Comment (saimarathore) [reply src/applications/SurvivalPredictor.h:887]: added comments

File: src/applications/SurvivalPredictor.h (hunk 1 above)
Comment (saimarathore) [reply src/applications/SurvivalPredictor.h:935]: these hard-coded values have been taken from the Matlab code. we may read these mapping from a csv in future to improve the code in the function.

File: src/applications/SurvivalPredictor.cpp [hunk 4]
```diff
@@ -352,69 +273,45 @@ VectorDouble SurvivalPredictor::CombineEstimates(const VariableLengthVectorType
 		float temp_abs, temp_pos1, temp_neg1, temp_1, temp_2;
 		// estimate for 1st vector
 		if (std::abs(estimates1[i]) < 2)
-		{
 			temp_abs = estimates1[i];
-		}
 		else
-		{
 			temp_abs = 0;
```
[the remark this answers is not in the document]
Comment (saimarathore) [reply src/applications/SurvivalPredictor.cpp:278]: +1/-1 are the class labels, and 2 is the capping limit that we apply on the predicted distances. taken directly from the matlab code.

File: src/common/CMakeLists.txt [hunk 5]
```diff
@@ -13,6 +13,7 @@ SET( ${PROJECT_NAME}_HDRS
 
 SET( ${PROJECT_NAME}_SRCS
   src/CaPTkGUIUtils.cpp
+  src/CaPTkutils.cpp
```
Comment (ashishsingh18) [inline src/common/CMakeLists.txt:16]: @saimarathore PR is failing. Did you forget the file that you added here?

Comment (saimarathore) [reply src/common/CMakeLists.txt]: this file was empty so i removed this line from CMakeLists

**Comment by ashishsingh18** [comment]
@CBICA/devteam This is ready for review.

**Pull request merged**